docs/src/main/asciidoc/_high-availability/Messaging.adoc

```diff
@@ -8,4 +8,460 @@ ifdef::env-github[]
+This exchange between the replicating server and secondary will block any journal related operations.
+The maximum length of time that this exchange will block is controlled by the `initial-replication-sync-timeout` configuration element.
+====
+
+Since replication will create a copy of the data at the secondary then in case of a successful fail-over, the secondary's data will be newer than the primary's data.
+If you configure your secondary to allow failback to the primary then when the primary is restarted it will be passive and the active secondary will synchronize its data with the passive primary before stopping to allow the passive primary to become active again.
+If both servers are shutdown then the administrator will have to determine which one has the latest data.
+
+[NOTE]
+.An Important Difference From Shared Store
+====
+If a shared-store secondary *does not* find a primary then it will just activate and service client requests like it is a primary.
+
+However, in the replication case, the secondary just keeps waiting for a primary to pair with because the secondary does not know whether its data is up-to-date.
+It cannot unilaterally decide to activate.
+To activate a replicating secondary using its current data the administrator must change its configuration to make it a primary server by changing `secondary` to `primary`.
+====
+
+==== Split Brain
+
+"Split Brain" is a potential issue that is important to understand.
+xref:Admin_Guide.html#network-isolation-split-brain[A whole chapter] has been devoted to explaining what it is and how it can be mitigated at a high level.
+
+==== Replication Configuration
+
+In a shared-store configuration brokers pair with each other based on their shared storage device.
+However, since replication configurations have no such shared storage device they must find each other another way.
+Servers can be grouped together explicitly using the same `group-name` in both the `primary` or the `secondary` elements.
+A secondary will only connect to a primary that shares the same node group name.
+
+[NOTE]
+.A `group-name` Example
+====
+Suppose you have 5 primary servers and 6 secondary servers:
+
+* `primary1`, `primary2`, `primary3`: with `group-name=fish`
+* `primary4`, `primary5`: with `group-name=bird`
+* `backup1`, `backup2`, `backup3`, `backup4`: with `group-name=fish`
+* `backup5`, `backup6`: with `group-name=bird`
+
+After joining the cluster the backups with `group-name=fish` will search for primary servers with `group-name=fish` to pair with.
+Since there is one secondary too many, the `fish` will remain with one spare secondary.
+
+The 2 backups with `group-name=bird` (`backup5` and `backup6`) will pair with primary servers `primary4` and `primary5`.
+====
+
+If `group-name` is not configured then the secondary will search for any primary that it can find the cluster.
```

Comment:
s/find the cluster/find in the cluster/g

I think ^^^ is the intended meaning.